RuddockWebsite/validation_utils.py

```diff
@@ -0,0 +1,124 @@
+import re
+import datetime
+from flask import flash, g
+from sqlalchemy import text
+from RuddockWebsite import constants
+
+def validate_username(username, flash_errors=True):
+  '''
+  Checks to make sure username is valid. If flash_errors is True, then the
+  errors will be displayed as flashes. Returns True if valid, else False.
+  '''
+  error = None
+  username_regex = re.compile(r'^[a-zA-Z0-9\-\_]+$')
```

Comment:
```python
username_regex = re.compile(r'\A[A-Za-z0-9_-]+\Z')
```
* Don't need to escape the `_` and `-` if you put `-` at the end.
* Instead of `^$` use `\A\Z` to match start and end of string only.